FIREFIT Data Models

rdinogit opened the pull request:
DIH2 FIREFIT experiment Data Models.

Commit:
FIREFIT Data Models

albertoabellagarcia reviewed (changes requested):
[DIH2/FIREFIT/HamInspectionBuffer/example/example-normalised.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-0ea4a585aa0340ce6d94168623bf4b3b) does not have the required property sorting. 

[DIH2/FIREFIT/LotInspectionBuffer/schema.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-d63b85f98fcdc94d8c0387b92fe48089) properties packages possibly needs to be an array. It cannot link to the very schema but to some part of the definitions (not provided in the schema) 

[DIH2/FIREFIT/ObjectInspectionResult/example/example-normalised.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-405f9412681a166b145f15877a6670e2) needs a type as required property. rename to example.json because it is a keyvalues example.


[DIH2/FIREFIT/Robot/examples/example-normalised.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-7c66620da9de1f2dd69925d0c78412ea)  needs a type as required property. rename to example.json because it is a keyvalues example.

[DIH2/FIREFIT/Sensor/examples/example-normalised.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-b1b681c4f1a2ed2f068bc0a1776f7000)  needs a type as required property. rename to example.json because it is a keyvalues example.

[DIH2/FIREFIT/SortingResultBuffer/examples/example-normalised.json](https://github.com/smart-data-models/incubated/pull/20/files#diff-ce5606fd2193bca243e1c30868bdd68f)  needs a type as required property. rename to example.json because it is a keyvalues example.

Commit:
Data models correction

Corrected the following issues:
- DIH2/FIREFIT/HamInspectionBuffer/example/example-normalised.json does not have the required property sorting.
- DIH2/FIREFIT/LotInspectionBuffer/schema.json properties packages possibly needs to be an array. It cannot link to the very schema but to some part of the definitions (not provided in the schema)
- DIH2/FIREFIT/ObjectInspectionResult/example/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.
- DIH2/FIREFIT/Robot/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.
- DIH2/FIREFIT/Sensor/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.
- DIH2/FIREFIT/SortingResultBuffer/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.

rdinogit commented:
Regarding the requested changes:

**DIH2/FIREFIT/HamInspectionBuffer/example/example-normalised.json does not have the required property sorting.**
There was a typo, the sorting property was not meant to exist it should be the property inspection instead.

**DIH2/FIREFIT/LotInspectionBuffer/schema.json properties packages possibly needs to be an array. It cannot link to the very schema but to some part of the definitions (not provided in the schema)**
This entity now does not reference to another. It is now an independent entity. Changes were made accordingly.

**DIH2/FIREFIT/ObjectInspectionResult/example/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.**
This entity was removed, considering the changes in the experiment and the *LotInspectionBuffer* entity update.

**DIH2/FIREFIT/Robot/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.**
Since this entity was generated automatically by the IoTAgent, the structure considered was the one presented by the IoTAgent. It was now changed to match the entity registered in the OrionCB as it should.

**DIH2/FIREFIT/Sensor/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.**
Since this entity was generated automatically by the IoTAgent, the structure considered was the one presented by the IoTAgent. It was now changed to match the entity registered in the OrionCB as it should.

**DIH2/FIREFIT/SortingResultBuffer/examples/example-normalised.json needs a type as required property. rename to example.json because it is a keyvalues example.**
*type* was present as a property and tagged as required. I was unable to detect what could be wrong. If the problem persists could you please elaborate?

All the renaming was done.

albertoabellagarcia commented:
There are some minor errors but I am going to accept it and to fix it because it takes more to explain than to do it
